src/client/libcurve_file.cpp

```diff
@@ -517,6 +531,37 @@ int FileClient::Recover(const std::string &filename, const UserInfo_t &userinfo,
     return -ret;
 }
 
+int FileClient::StatFile(int fd, FileStatInfo *finfo) {
+    {
+        ReadLockGuard lk(rwlock_);
+        auto iter = fileserviceMap_.find(fd);
+        if (iter == fileserviceMap_.end()) {
+            LOG(ERROR) << "StatFile failed not found fd = " << fd;
+            return -LIBCURVE_ERROR::FAILED;
+        }
+    }
+
+    FileInstance *instance = fileserviceMap_[fd];
+
+    FInfo_t fi = instance->GetCurrentFileInfo();
+    finfo->id = fi.id;
+    finfo->parentid = fi.parentid;
+    finfo->ctime = fi.ctime;
+    finfo->length = fi.length;
+    finfo->filetype = fi.filetype;
+    finfo->stripeUnit = fi.stripeUnit;
+    finfo->stripeCount = fi.stripeCount;
+
+    memcpy(finfo->filename, fi.filename.c_str(),
+               std::min(sizeof(finfo->filename), fi.filename.size() + 1));
+    memcpy(finfo->owner, fi.owner.c_str(),
+            std::min(sizeof(finfo->owner), fi.owner.size() + 1));
+
+    finfo->fileStatus = static_cast<int>(fi.filestatus);
```

Comment:
consider extract into a function, because L580-593 has the same logic

Reply:
done